app/lib/humane_transport_app.dart

```diff
@@ -11,6 +12,12 @@ class HumaneTransportApp extends StatelessWidget {
       providers: [
         ChangeNotifierProvider<NavigationProvider>(
             create: (context) => NavigationProvider()),
+        ChangeNotifierProvider<AuthenticationProvider>(
+          create: (context) => AuthenticationProvider(),
+        ),
+        StreamProvider(
```

Comment:
Why do we need a stream provider on top?

Reply:
If it's not on top, then widgets above won't be able to listen to auth changes changes. The provider must and should always be above the widget who needs it. In this case, our Auth wrapper needs it.

Reply:
What do you mean by widgets above?

Reply:
Hard to explain without some kind of widget tree diagram to show this. But I believe stream works similar to [provider](https://flutter.dev/docs/development/data-and-backend/state-mgmt/simple). The article can explain it better and also shows a widget diagram